src/User/UserContainer.jsx

```diff
@@ -10,35 +10,72 @@ const UserContainer = () => {
   const [user, setUser] = useState(null);
   const [isLoading, setIsLoading] = useState(true);
   const [error, setError] = useState(false);
+  const [isCurrentUser, setIsCurrentUser] = useState(false);
+  const [managerName, setManagerName] = useState("");
   const { id } = useParams();
   const [cookies] = useCookies([cookieName]);
 
+  const checkIfCurrentUser = (fetchedId, currentUserId) => {
+    if (Number(fetchedId) === currentUserId) {
+      setIsCurrentUser(true);
+    }
+  };
+
+  const getManagerName = async managerId => {
+    if (managerId) {
+      try {
+        const manager = await getUserById(
+          managerId,
+          cookies.sessionToken.token
+        );
+        setManagerName(`${manager.firstName} ${manager.lastName}`);
+      } catch (error) {
+        setError(true);
+      }
+    } else {
+      setManagerName("None Assigned");
+    }
+  };
+
   const fetchData = async userId => {
     setIsLoading(true);
 
     try {
       const user = await getUserById(userId, cookies.sessionToken.token);
       setUser(user);
+      checkIfCurrentUser(userId, cookies.sessionToken.userId);
     } catch (error) {
       setError(true);
     }
+
     setIsLoading(false);
   };
 
   useEffect(() => {
     if (id) {
       fetchData(id);
     }
+    if (isCurrentUser) {
+      getManagerName(user.managerId);
```

Comment:
this also therefore relies on user.managerId, should probably be a separate useEffect?